example/lib/screens/message_view.dart

```diff
@@ -2,33 +2,60 @@ import 'package:flutter/material.dart';
 import 'package:airship/airship.dart';
 import 'package:airship_example/styles.dart';
 
-class MessageView extends StatelessWidget {
-  final InboxMessage message;
+class MessageView extends StatefulWidget {
+  final String messageId;
 
-  MessageView({this.message});
+  MessageView({this.messageId});
+
+  @override
+  _MessageViewState createState() => _MessageViewState();
+}
+
+class _MessageViewState extends State<MessageView>  {
+  bool isLoading = true;
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-        appBar: AppBar(
-          title: Text("${message.title}"),
-          backgroundColor: Styles.background,
-        ),
-        body: Container(
-          child: InboxMessageView(onViewCreated:onInboxMessageViewCreated),
-        )
-    );
+    return FutureBuilder<List<InboxMessage>>(
+        future: Airship.inboxMessages,
+        builder: (context, snapshot) {
+          InboxMessage message;
+
+          List<InboxMessage> list = [];
+
+          if (snapshot.hasData) {
+            list = List<InboxMessage>.from(snapshot.data);
+          }
+
+          message = list.firstWhere((thisMessage) =>
+          widget.messageId == thisMessage.messageId,
+              orElse: () => null);
+
+          return Scaffold(
+            appBar: AppBar(
+              title: message != null ? Text("${message.title}") : Container(),
```

Comment:
Is the view empty if the message is nil? Is there an alert like on iOS? Does it just show the inbox?

Reply:
The view body is empty until the web page request completes. The message property here is null until the message list future returns. In this case, we just show an invisible container until the title is ready to be displayed.